### ISSUE TITLE
Get relationship from notification

### ISSUE THREAD

**Issue by g5122053** [laravel/framework#15693]
Is it possible to retrieve relationships given a notification `id`?

For example, using this snippet from the [Laravel docs](https://laravel.com/docs/5.3/notifications#database-notifications):

```
/**
 * Get the array representation of the notification.
 *
 * @param  mixed  $notifiable
 * @return array
 */
public function toArray($notifiable)
{
    return [
        'invoice_id' => $this->invoice->id,
        'amount' => $this->invoice->amount,
    ];
}
```

Assuming there exists an `Invoice` model and table, how can I retrieve the `Invoice` record by using the `invoice_id` value in the notification data when getting user notifications:

```
$user = App\User::find(1);

foreach ($user->notifications as $notification) {
    //
}
```

**Comment by themsaid** [comment]
`$notification->toArray()['invoice_id']`

**Comment by g5122053** [comment]
I'm not sure how that helps.

First, I get the following error when I print that:

`Undefined index: invoice_id`

Second, that would only get me the `invoice_id`, but not the actual `Invoice` record relationship.

**Comment by g5122053** [comment]
I'll give a better example.

Let's say I want to notify a user about a new post that I've made.

I don't want to store the entire `Post` record in the notification, because the post contents can be edited (meaning that if I edited the post, the user should be able to see those edits in the notification). So instead, I just want to store the `id` of the `Post` so that I can later use that to retrieve the `Post` record from the `posts` table.

How would I do this?
